### ISSUE TITLE
make[2]: *** [src/modules/perlcore/CMakeFiles/kviperlcore.dir/build.make:160: src/modules/perlcore/libkviperlcore.so] Error 1

### ISSUE THREAD

**Issue by jaxson5** [kvirc/KVIrc#2600]
make fails on the perl module
[ 85%] Building CXX object src/modules/perlcore/CMakeFiles/kviperlcore.dir/kviperlcore_autogen/mocs_compilation.cpp.o
[ 85%] Building CXX object src/modules/perlcore/CMakeFiles/kviperlcore.dir/libkviperlcore.cpp.o
/home/jax5/temp/KVIrc/src/modules/perlcore/libkviperlcore.cpp:83: warning: "dNOOP" redefined
   83 | #define dNOOP (void)0
      | 
In file included from /home/jax5/temp/KVIrc/src/modules/perlcore/libkviperlcore.cpp:68:
/usr/lib/x86_64-linux-gnu/perl/5.36/CORE/perl.h:640: note: this is the location of the previous definition
  640 | #define dNOOP struct Perl___notused_struct
      | 
[ 86%] Linking CXX shared module libkviperlcore.so
/usr/bin/ld: cannot find -lperl: No such file or directory
collect2: error: ld returned 1 exit status
make[2]: *** [src/modules/perlcore/CMakeFiles/kviperlcore.dir/build.make:160: src/modules/perlcore/libkviperlcore.so] Error 1
make[1]: *** [CMakeFiles/Makefile2:7125: src/modules/perlcore/CMakeFiles/kviperlcore.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

KVIrc:
KVIrc 5.2.0 tar.gz (grabbed the latest git and cloned it - modified with bug fix #2598)

Debian:
Debian GNU/Linux 12.2.0-14

uname:
Linux main 6.1.0-13-amd64 https://github.com/kvirc/KVIrc/issues/1 SMP PREEMPT_DYNAMIC Debian 6.1.55-1 (2023-09-29) x86_64 GNU/Linux

cmake:
cmake version 3.25.1

qmake:
QMake version 3.1
Using Qt version 6.4.2 in /usr/lib/x86_64-linux-gnu

gcc:
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/12/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none:amdgcn-amdhsa
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Debian 12.2.0-14' --with-bugurl=file:///usr/share/doc/gcc-12/README.Bugs --enable-languages=c,ada,c++,go,d,fortran,objc,obj-c++,m2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-12 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --enable-libphobos-checking=release --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --enable-cet --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-12-bTRWOB/gcc-12-12.2.0/debian/tmp-nvptx/usr,amdgcn-amdhsa=/build/gcc-12-bTRWOB/gcc-12-12.2.0/debian/tmp-gcn/usr --enable-offload-defaulted --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 12.2.0 (Debian 12.2.0-14)

Attached:
compile logKVIrc:
KVIrc 5.2.0 tar.gz (grabbed the latest git and cloned it)

Debian:
Debian GNU/Linux 12.2.0-14

uname:
Linux main 6.1.0-13-amd64 https://github.com/kvirc/KVIrc/issues/1 SMP PREEMPT_DYNAMIC Debian 6.1.55-1 (2023-09-29) x86_64 GNU/Linux

cmake:
cmake version 3.25.1

qmake:
QMake version 3.1
Using Qt version 6.4.2 in /usr/lib/x86_64-linux-gnu

gcc:
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/12/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none:amdgcn-amdhsa
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Debian 12.2.0-14' --with-bugurl=file:///usr/share/doc/gcc-12/README.Bugs --enable-languages=c,ada,c++,go,d,fortran,objc,obj-c++,m2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-12 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --enable-libphobos-checking=release --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --enable-cet --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-12-bTRWOB/gcc-12-12.2.0/debian/tmp-nvptx/usr,amdgcn-amdhsa=/build/gcc-12-bTRWOB/gcc-12-12.2.0/debian/tmp-gcn/usr --enable-offload-defaulted --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 12.2.0 (Debian 12.2.0-14)

Attached:
compile log
[compile.txt](https://github.com/kvirc/KVIrc/files/14075448/compile.txt)

**Comment by ctrlaltca** [comment]
Looks like you actually have perl installed on your system, but not the perl library itself (needed to link external programs with perl).
If you run `perl -MExtUtils::Embed -e ldopts`, it should return something like:
`-Wl,-E -Wl,-rpath,/usr/lib64/perl5/CORE  -fstack-protector-strong -L/usr/local/lib  -L/usr/lib64/perl5/CORE -lperl -lpthread -ldl -lm -lcrypt -lutil -lc`

These are the options that KVirc needs to link with perl. 
I've tried to have a look at Debian packages, and it seems that perl requires libperl as expected (https://packages.debian.org/bookworm/perl), and libperl provides the required library (https://packages.debian.org/bookworm/amd64/libperl5.36/filelist has libperl.so.5.36 in it). Still, since we are linking against a non-versioned -lperl, you need to have libperl.so that is provided by the libperl-dev package (https://packages.debian.org/bookworm/amd64/libperl-dev/filelist).

tl;dr: `apt install libperl-dev`

**Comment by jaxson5** [comment]
Hi ctrlaltca,
That works fine. I did reinstall the perl devs but must have missed that one.
Thanks again.
jax
